owltail-cognition/pyPodcastParser
# IHRACP-5834: Pypodcaster repo gha

## JIRA Ticket
- [IHRACP-5834](https://ihm-it.atlassian.net/browse/IHRACP-5834)

## Description :

### CI/CD Workflow and Versioning Explanation
- CI/CD Pipeline Overview:

- This CI/CD workflow automates the deployment of the pypodcastparser-ihr Python package to AWS CodeArtifact. It dynamically determines the version based on the branch, using the latest release tag and commit count.
- When changes are pushed to any branch, the CI/CD pipeline is triggered, resulting in the creation of a new build/image/package on CodeArtifact. The naming convention for the development versions follows the format: `current_release_version.dev.<commit-number>`. Examples include `1.5.1.dev1`, `1.5.1.dev2`, and so on.
- and for master branch it will craete with lattest release tag `1.6.1`


#### Major Changes:
- Create CI/CD to automatically deploy to CodeArtifact

## How to test:
-  push any changes in any branch and check on CodeArtifact has created new build and expected result is below
`current_release_version.dev.<commit-id>`
 Examples:  `1.5.1.dev1`
- if your on master branch or relsesed new version then you will get `1.6.1`
